runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -360,9 +450,28 @@ struct NullQubit final : public Catalyst::Runtime::QuantumDevice {
         return {0, 0, 0, {}, {}};
     }
 
+    /**
+     * @brief Returns the number of gates used since the last time all qubits were released. Only
+     * works if resource tracking is enabled
+     */
+    std::size_t ResourcesGetNumGates() { return resource_data_["num_gates"]; }
```

Comment:
Could you use the tailing signature everywhere?

I mean, 
```suggestion
    auto ResourcesGetNumGates() -> std::size_t { return resource_data_["num_gates"]; }
```